cloud/scope/cluster.go

```diff
@@ -46,22 +46,22 @@ func NewClusterScope(params ClusterScopeParams) (*ClusterScope, error) {
 	if params.Cluster == nil {
 		return nil, errors.New("failed to generate new scope from nil Cluster")
 	}
-	if params.AzureCluster == nil {
-		return nil, errors.New("failed to generate new scope from nil AzureCluster")
+	if params.AzureStackEdgeCluster == nil {
+		return nil, errors.New("failed to generate new scope from nil AzureStackEdgeCluster")
 	}
 
 	if params.Logger == nil {
 		params.Logger = klogr.New()
 	}
 
 	log := klogr.New()
-	err := params.AzureClients.setDBECredentials(params.AzureCluster.Spec.SubscriptionID)
+	err := params.AzureClients.setDBECredentials(params.AzureStackEdgeCluster.Spec.SubscriptionID)
 	if err != nil {
 		log.Info(err.Error())
 		return nil, errors.Wrap(err, "failed to create Azure session")
```

Comment:
Is there any retry which acquiring ARM token, this is to handle some network issue

Reply:
There is no retry in acquiring the ARM token but newClusterScope is called for every reconcile iteration (newClusterScope calls setAzureStackEdgeCredentials), so if there is any network issue in acquiring the ARM token, it will be called again at the next reconcile iteration.